pkg/liqo-controller-manager/foreign-cluster-operator/foreign-cluster-controller.go

```diff
@@ -392,10 +401,28 @@ func (r *ForeignClusterReconciler) SetupWithManager(mgr ctrl.Manager, workers ui
 		Owns(&netv1alpha1.TunnelEndpoint{}).
 		Watches(&source.Kind{Type: &corev1.Secret{}}, authenticationtoken.GetAuthTokenSecretEventHandler(r.Client),
 			builder.WithPredicates(authenticationtoken.GetAuthTokenSecretPredicate())).
+		Watches(&source.Kind{Type: &sharingv1alpha1.ResourceOffer{}},
+			handler.EnqueueRequestsFromMapFunc(r.resourceOfferHandler)).
 		WithOptions(controller.Options{MaxConcurrentReconciles: int(workers)}).
 		Complete(r)
 }
 
+func (r *ForeignClusterReconciler) resourceOfferHandler(obj client.Object) []ctrl.Request {
+	ro := obj.(*sharingv1alpha1.ResourceOffer)
+	var clusterID string
+	if _, ok := ro.Labels[liqoconst.ReplicationStatusLabel]; ok { // incoming resource offer
+		clusterID = ro.Labels[liqoconst.ReplicationOriginLabel]
+	} else {
+		clusterID = ro.Labels[liqoconst.ReplicationDestinationLabel]
+	}
+	remoteCluster, err := foreignclusterutils.GetForeignClusterByID(context.Background(), r.Client, clusterID)
+	if err != nil {
+		klog.Warningf("Could not handle resource offer update: %s", err)
```

Comment:
```suggestion
		klog.Warningf("Could not handle resource offer %q update: %s", klog.KObj(ro), err)
```

this can make easier to understand which resource offer is causing the problem